tests/integration/server/api/users/index.get.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it, vi, afterEach } from "vitest"
+import { setup, $fetch } from "@nuxt/test-utils/e2e"
+
+describe("Api Route /api/users", async () => {
+  await setup()
+  afterEach(() => {
+    vi.resetAllMocks()
+  })
+
+  it("Should return a list of users", async () => {
+    const response = await $fetch("/api/users")
+    // Expected output structure
+    const expectedStructure = [
+      { username: "Admin", role: "ADMIN", verified: true },
+      { username: "Moderator", role: "MODERATOR", verified: true },
+      { username: "User", role: "USER", verified: true },
+    ]
+
+    // Ensure data has length of at least 3 (migration with seeds work)
+    expect(response.length).toBeGreaterThan(3)
+
+    // Expect all listed elements
+    expectedStructure.forEach((expectedItem) => {
+      expect(response).toContainEqual(expect.objectContaining(expectedItem))
+    })
+    response.forEach((item) => {
+      // Expect a UUID for each item
+      expect(item).toHaveProperty("id")
+      // Expect the list items to have a name
+      expect(item).toHaveProperty("username")
+      // Expect the list items to have a role
+      expect(item).toHaveProperty("role")
```

Comment:
Remove all the expect(item).toHaveProperty tests since they are then covered by the type check above